IntegrationTests/common/script/CompareMemPrintsFW.py

```diff
@@ -182,31 +221,105 @@ def compare(comparison_filename="", fail_on_error=False, file_location='./', pre
-
-    for i in range(17,25):
-        if os.path.exists(('./dataOut/VMPROJ_L3PHIC%i.txt' % (i))):
-            ret_sum += compare(comparison_filename=("./dataOut/VMPROJ_L3PHIC%i.txt" % (i)), fail_on_error=False, file_location=args.file_location, predefined=args.predefined,
-                               reference_filenames=[("../../../emData/MemPrints/VMProjections/VMProjections_VMPROJ_L3PHIC%i_04.dat" % (i))], save=args.save, verbose=args.verbose)
 
-    for i in range(17,25):
-        if os.path.exists(('./dataOut/CM_L3PHIC%i.txt' % (i))):
-            ret_sum += compare(comparison_filename=("./dataOut/CM_L3PHIC%i.txt" % (i)), fail_on_error=False, file_location=args.file_location, predefined=args.predefined,
-                               reference_filenames=[("../../../emData/MemPrints/Matches/CandidateMatches_CM_L3PHIC%i_04.dat" % (i))], save=args.save, verbose=args.verbose)
+    if (args.predefined == "PRMEMC"):
+        if os.path.exists('./dataOut/AP_L3PHIC.txt'):
+            ret_sum += compare(comparison_filename="./dataOut/AP_L3PHIC.txt", fail_on_error=False, file_location=args.file_location, predefined=args.predefined,
+                               reference_filenames=["../../../emData/MemPrints/TrackletProjections/AllProj_AP_L3PHIC_04.dat"], save=args.save, verbose=args.verbose)
+        
+        for i in range(17,25):
+            if os.path.exists(('./dataOut/VMPROJ_L3PHIC%i.txt' % (i))):
+                ret_sum += compare(comparison_filename=("./dataOut/VMPROJ_L3PHIC%i.txt" % (i)), fail_on_error=False, file_location=args.file_location, predefined=args.predefined,
+                                   reference_filenames=[("../../../emData/MemPrints/VMProjections/VMProjections_VMPROJ_L3PHIC%i_04.dat" % (i))], save=args.save, verbose=args.verbose)
+        
+        for i in range(17,25):
+            if os.path.exists(('./dataOut/CM_L3PHIC%i.txt' % (i))):
+                ret_sum += compare(comparison_filename=("./dataOut/CM_L3PHIC%i.txt" % (i)), fail_on_error=False, file_location=args.file_location, predefined=args.predefined,
+                                   reference_filenames=[("../../../emData/MemPrints/Matches/CandidateMatches_CM_L3PHIC%i_04.dat" % (i))], save=args.save, verbose=args.verbose)
+        
+        ret_sum += compare(comparison_filename="./dataOut/FM_L1L2_L3PHIC.txt", fail_on_error=False, file_location=args.file_location, predefined=args.predefined,
+                           reference_filenames=["../../../emData/MemPrints/Matches/FullMatches_FM_L1L2_L3PHIC_04.dat"], save=args.save, verbose=args.verbose)
+        ret_sum += compare(comparison_filename="./dataOut/FM_L5L6_L3PHIC.txt", fail_on_error=False, file_location=args.file_location, predefined=args.predefined,
+                           reference_filenames=["../../../emData/MemPrints/Matches/FullMatches_FM_L5L6_L3PHIC_04.dat"], save=args.save, verbose=args.verbose)
+
+    elif (args.predefined == "IRVMR"):
```

Comment:
I assume all this hand-written code will need rewriting whenever we include different modules in the IRVMR chain or PRMEMC chain etc. Not required now, but we should think how to automate this.

Reply:
I automated this.

Reply:
clever